File: #3068.find-the-maximum-sum-of-node-values.cpp
class Solution {
public:
    long long maximumValueSum(vector<int>& nums, int k, vector<vector<int>>& edges) {

        long long sum = 0;
        long long cnt = 0;
        for (auto &i : nums) {
            if ((i ^ k) > i) {
                cnt++;
                i ^= k;
            }
            sum += i;
        }

        if (cnt & 1) {
            int min = 1e9;
            for (auto i : nums) {
                min = std::min(min, i - (i ^ k));
            }
            sum -= min;
        }
        return sum;
    }
};

File: #3068.find-the-maximum-sum-of-node-values.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @param {number[][]} edges
 * @return {number}
 */
var maximumValueSum = function(nums, k, edges) {
    const n = nums.length;
    const temp = Array.from({ length: n }, () => [-1, -1]);
    function calculateMax(curInd, isEven) {
        if (curInd === n) {
            return isEven === 1 ? 0 : -Infinity;
        }
        if (temp[curInd][isEven] !== -1) {
            return temp[curInd][isEven];
        }

        const noXor = nums[curInd] + calculateMax(curInd + 1, isEven);
        const withXor = (nums[curInd] ^ k) + calculateMax(curInd + 1, 1 - isEven);
        const mxPossible = Math.max(noXor, withXor);
        temp[curInd][isEven] = mxPossible;
        return mxPossible;
    }
    return calculateMax(0, 1);
};
